app/services/puppet_class_importer.rb

```diff
@@ -182,6 +182,13 @@ def old_environments
     db_environments - actual_environments
   end
 
+  def current_user_environments
+    ids = [Organization.my_organizations, Location.my_locations].flatten.map do |taxonomy|
+      taxonomy.environment_ids
+    end
+    Environment.where(:id => ids.reduce(:+).uniq).map(&:to_s)
```

Comment:
If you use authorized scope, please always specify permission which you want user to be authorized for.